src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/config/AuditConfig.java

```diff
@@ -0,0 +1,268 @@
+            final boolean logRequestBody = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_LOG_REQUEST_BODY, true);
+            final boolean resolveIndices = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_RESOLVE_INDICES, true);
+            final boolean excludeSensitiveHeaders = settings.getAsBoolean(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_EXCLUDE_SENSITIVE_HEADERS, true);
+
+            final EnumSet<AuditCategory> disabledRestCategories = AuditCategory.parse(getSettingAsSet(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_REST_CATEGORIES,
+                    DEFAULT_DISABLED_CATEGORIES,
+                    true));
+
+            final EnumSet<AuditCategory> disabledTransportCategories = AuditCategory.parse(getSettingAsSet(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_DISABLED_TRANSPORT_CATEGORIES,
+                    DEFAULT_DISABLED_CATEGORIES,
+                    true));
+
+            final Set<String> ignoredAuditUsers = getSettingAsSet(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false);
+
+            final Set<String> ignoredComplianceUsersForRead = getSettingAsSet(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false);
+
+            final Set<String> ignoredComplianceUsersForWrite = getSettingAsSet(
+                    settings,
+                    ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_IGNORE_USERS,
+                    DEFAULT_IGNORED_USERS,
+                    false);
+
+            final Set<String> ignoreAuditRequests = ImmutableSet.copyOf(settings.getAsList(
+                    ConfigConstants.OPENDISTRO_SECURITY_AUDIT_IGNORE_REQUESTS,
+                    Collections.emptyList()));
+
+            return new Filter(isRestApiAuditEnabled,
+                    isTransportAuditEnabled,
+                    resolveBulkRequests,
+                    logRequestBody,
+                    resolveIndices,
+                    excludeSensitiveHeaders,
+                    ignoredAuditUsers,
+                    ignoredComplianceUsersForRead,
+                    ignoredComplianceUsersForWrite,
+                    ignoreAuditRequests,
+                    disabledRestCategories,
+                    disabledTransportCategories);
+        }
+
+        private static Set<String> getSettingAsSet(final Settings settings, final String key, final List<String> defaultList, final boolean ignoreCaseForNone) {
+            final List<String> list = settings.getAsList(key, defaultList);
+            if (list.size() == 1 && "NONE".equals(ignoreCaseForNone? list.get(0).toUpperCase() : list.get(0))) {
```

Comment:
In case conversion to `Set` is done inside this function, it is better to add `isEmpty()` check here. I am not 100% convinced that it is necessary to convert to `Set` from `List` here as it is not always necessary and in the future there may be different requirements for case-sensitivity of the returned `Set`.

Reply:
Will change it when needed in the future.